packages/codegen-ui-golden-files/lib/react/forms/datastore-create-form-with-many-to-many-relationship/TagCreateForm.jsx

```diff
@@ -0,0 +1,399 @@
+  );
+}
+export default function TagCreateForm(props) {
+  const {
+    clearOnSuccess = true,
+    onSuccess,
+    onError,
+    onSubmit,
+    onCancel,
+    onValidate,
+    onChange,
+    overrides,
+    ...rest
+  } = props;
+  const initialValues = {
+    label: '',
+    Posts: [],
+  };
+  const [label, setLabel] = React.useState(initialValues.label);
+  const [Posts, setPosts] = React.useState(initialValues.Posts);
+
+  const postRecords = useDataStoreBinding({
+    type: 'collection',
+    model: Post,
+  }).items;
+
+  // generated from valueMappings config
+  const getDisplayValue = {
+    Posts: (record) => record?.title,
+  };
+
+  const [currentPostsValue, setCurrentPostsValue] = React.useState(undefined);
+  const [currentPostsDisplayValue, setCurrentPostsDisplayValue] = React.useState('');
+
+  const [errors, setErrors] = React.useState({});
+  const resetStateValues = () => {
+    setLabel(initialValues.label);
+    setPosts(initialValues.Posts);
+    setCurrentPostsValue(undefined);
+    setCurrentPostsDisplayValue('');
+    setErrors({});
+  };
+
+  const PostsRef = React.createRef();
+
+  const validations = {
+    label: [],
+    Posts: [],
+  };
+
+  // new arg so we're validating string, not Model
+  const runValidationTasks = async (fieldName, currentValue, getDisplayValue) => {
+    const value = getDisplayValue ? getDisplayValue(currentValue) : currentValue;
+    let validationResponse = validateField(value, validations[fieldName]);
+    const customValidator = fetchByPath(onValidate, fieldName);
+    if (customValidator) {
+      validationResponse = await customValidator(value, validationResponse);
+    }
+    setErrors((errors) => ({ ...errors, [fieldName]: validationResponse }));
+    return validationResponse;
+  };
+  return (
+    <Grid
+      as="form"
+      rowGap="15px"
+      columnGap="15px"
+      padding="20px"
+      onSubmit={async (event) => {
+        event.preventDefault();
+        let modelFields = {
+          label,
+          Posts,
+        };
+        const validationResponses = await Promise.all(
+          Object.keys(validations).reduce((promises, fieldName) => {
+            if (Array.isArray(modelFields[fieldName])) {
+              promises.push(
+                ...modelFields[fieldName].map((item) =>
+                  // pass in getDisplayValue callback to validate string, not Model
+                  runValidationTasks(fieldName, item, getDisplayValue[fieldName]),
+                ),
+              );
+              return promises;
+            }
+            promises.push(runValidationTasks(fieldName, modelFields[fieldName], getDisplayValue[fieldName]));
+            return promises;
+          }, []),
+        );
+        if (validationResponses.some((r) => r.hasError)) {
+          return;
+        }
+        if (onSubmit) {
+          modelFields = onSubmit(modelFields);
+        }
+        try {
+          const tag = await DataStore.save(new Tag(modelFields));
```

Comment:
I'm guessing you tried this out locally, but does save fail or anything if you try to pass in `Post`s to the `post` field like this?

Reply:
You can just resolve this. Figure you'll address it while writing E2E tests.